### DISCUSSION TITLE
GraphQL: Get file content for a specific file in each repo

### DISCUSSION THREAD

**Discussion by rajbos** [community/community#32469] [Apps, API and Webhooks] [answered]
I'm trying to get the data I want from an organization in one go and can't figure out how to filter the objects down to the two filenames I want: `action.yml` and `action.yaml`.

This is what I got so far:
````
query {
  organization(login:"actions") {
    repositories(first: 10) {
      nodes{
        name
        pushedAt
        object(expression: "HEAD:"){
          ... on Tree {
           entries {
            name
            type
            object {
              ... on Blob {
                byteSize
                text                
              }
            }
          } 
          }
        }
      }
    }
  }
}
````

**Comment by byrneh** [comment, marked as answer]
Hi @rajbos, I believe this is the answer you are looking for.

query {
  organization(login:"actions") {
    repositories(first: 10) {
      nodes{
        name
        content_a: object(expression: "HEAD:action.yml"){
          ... on Blob {
            text
            byteSize
          }
        }
        content_b: object(expression: "HEAD:action.yaml"){
          ... on Blob {
            text
            byteSize
          }
        }
      }
    }
  }
}

**Reply by ghostinhershell** [reply]
Hey @rajbos 👋

Did @byrneh's solution here answer your question?

**Reply by rajbos** [reply]
Yes it did! Thanks

**Reply by yc1838** [reply]
Hey @byrneh thanks for your info it was very helpful. I wonder if there is a way to get back files that end with .yml instead of pulling the file only with specific names?

**Reply by maciekbanas** [reply]
Hi, thanks for this thread, it helps a lot. Do you know how to put a `default` branch in the expression, so it works, instead of `master`, `main` or `HEAD`?

**Reply by Shibayan95** [reply]
@maciekbanas You first need to fetch the default branch def , then again make a query

**Reply by maciekbanas** [reply]
Thanks @Shibayan95 I had some hope it could go other way, as on GitLab, where you can define the branch in the file query.
